shell/src/main/java/alluxio/cli/fs/command/LsPathInAlluxio.java

```diff
@@ -0,0 +1,130 @@
+     *
+     * @param path The {@link AlluxioURI} path as the input of the command
+     * @param readable whether to print info of human readable format
+     */
+    private void lsPathInAlluxio(AlluxioURI path, boolean readable) throws AlluxioException, IOException {
+
+        Timer timer = new Timer();
+        timer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                System.out.println("Getting directory status of %s files or sub-directories "
+                        + "may take a while.");
+            }
+        }, 10000);
+        ListStatusPOptions.Builder optionsBuilder = ListStatusPOptions.newBuilder();
+        optionsBuilder.setRecursive(true);
+        List<URIStatus> statuses = mFileSystem.listStatus(path, optionsBuilder.build());
+        timer.cancel();
+
+        Map<String, Long> distributionMap = new HashMap<>();
+        for (URIStatus status: statuses) {
+            for (FileBlockInfo fileBlockInfo : status.getFileBlockInfos()) {
+                Long length = fileBlockInfo.getBlockInfo().getLength();
+                for (BlockLocation blockLocation : fileBlockInfo.getBlockInfo().getLocations()) {
+                    distributionMap.put(blockLocation.getWorkerAddress().getHost(),
+                        distributionMap.getOrDefault(blockLocation.getWorkerAddress().getHost(), 0L) + length);
+                }
+            }
+        }
+
+        distributionMap.forEach((workerHostName, size) -> {
+            printInfo(workerHostName, readable ? FormatUtils.getSizeFromBytes(size) : String.valueOf(size));
+        });
+    }
+
+    /**
+     * Prints the size messages.
+     *
+     * @param workerHostName Host name of the worker
+     * @param inAlluxioMessage the in Alluxio size message to print
+     */
+    private static void printInfo(String workerHostName, String inAlluxioMessage) {
+        System.out.println(String.format("%-25s %s", workerHostName, inAlluxioMessage));
+    }
+
+    @Override
+    protected void processHeader(CommandLine cl) {
+        printInfo("Worker Host Name", "In Alluxio");
+    }
+
+    @Override
+    protected void runPlainPath(AlluxioURI path, CommandLine cl)
+            throws AlluxioException, IOException {
+        lsPathInAlluxio(path, cl.hasOption(READABLE_OPTION_NAME));
+    }
+
+    @Override
+    public int run(CommandLine cl) throws AlluxioException, IOException {
+        String[] args = cl.getArgs();
+        AlluxioURI path = new AlluxioURI(args[0]);
+        runWildCardCmd(path, cl);
+        return 0;
+    }
+
+    @Override
+    public String getCommandName() {
+        return "lsPathInAlluxio";
+    }
+
+    @Override
+    public String getUsage() {
+        return "lsPathInAlluxio [-h] <path>";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Displays information for In-Alluxio data size under the path group by worker.";
```

Comment:
Maybe `Displays information for in-Alluxio data size under the path, grouped by worker` ?

Reply:
Ok